blosc/directories.c

```diff
@@ -49,86 +67,220 @@
 
       ret = remove(fname);
       free(fname);
+      if (ret < 0) {
+        BLOSC_TRACE_ERROR("Could not remove file %s", fname);
+        _findclose(file);
+        return BLOSC2_ERROR_FAILURE;
+      }
+    }
+
+    rmdir(dir_path);
+    _findclose(file);
+    return 0;
+  }
+
+  int blosc2_rename_dir(const char* dir_path, char* new_dir_path) {
+    // When searching inside the directory path has to end with '*'
+    char* search_path;
+    char last_char = dir_path[strlen(dir_path) - 1];
+    if (last_char != '\\' || last_char != '/') {
+      search_path = malloc(strlen(dir_path) + 2 + 1);
+      sprintf(search_path, "%s\\*", dir_path);
+    }
+    else {
+      search_path = malloc(strlen(dir_path) + 1 + 1);
+      strcpy(search_path, dir_path);
+      strcat(search_path, "*");
+    }
+    char* path = blosc2_normalize_dirpath(dir_path);
+    char* new_path = blosc2_normalize_dirpath(new_dir_path);
+    char* fname;
+    char* new_fname;
+
+    struct _finddata_t cfile;
+    intptr_t file = _findfirst(search_path, &cfile);
+    free(search_path);
+
+    if (file == -1) {
+      BLOSC_TRACE_ERROR("Could not open the file.");
+      return BLOSC2_ERROR_FILE_OPEN;
+    }
+    int ret;
+    if (_mkdir(new_path) < 0 ){
+      free(path);
+      free(new_path);
+      BLOSC_TRACE_ERROR("Could not create the new directory. ");
+      return BLOSC2_ERROR_FAILURE;
+    }
+    while ( _findnext(file, &cfile) == 0) {
+      if (strcmp(".", cfile.name) == 0 || strcmp("..", cfile.name) == 0) {
+        continue;
+      }
+      fname = malloc(strlen(path) + strlen(cfile.name) + 1);
+      sprintf(fname, "%s%s", path, cfile.name);
+
+      new_fname = malloc(strlen(new_path) + strlen(cfile.name) + 1);
+      sprintf(new_fname, "%s%s", new_path, cfile.name);
+
+      ret = rename(fname, new_fname);
```

Comment:
I think rename accepts dir entries besides files (at least on UNIX).  That would be single operation for the whole directory instead of many.  Can you have a try at it?

Reply:
You were right, even on Windows.